platform/windows/os_windows.cpp

```diff
@@ -915,7 +915,25 @@ LRESULT OS_Windows::WndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
 			}
 
 		} break;
-
+		case WM_NCHITTEST: {
+			// Only handle the hit test if the left mouse button is pressed.
+			bool buttons_swapped = GetSystemMetrics(SM_SWAPBUTTON);
+			if (!GetAsyncKeyState(VK_LBUTTON) && !buttons_swapped || !GetAsyncKeyState(VK_RBUTTON) && buttons_swapped) break;
+
+			switch (drag_mode) {
+				case DRAG_MODE_MOVE: return HTCAPTION;
+				case DRAG_MODE_RESIZE_TOP: return HTTOP;
+				case DRAG_MODE_RESIZE_RIGHT: return HTRIGHT;
+				case DRAG_MODE_RESIZE_BOTTOM: return HTBOTTOM;
+				case DRAG_MODE_RESIZE_LEFT: return HTLEFT;
+				case DRAG_MODE_RESIZE_TOPLEFT: return HTTOPLEFT;
+				case DRAG_MODE_RESIZE_TOPRIGHT: return HTTOPRIGHT;
+				case DRAG_MODE_RESIZE_BOTTOMRIGHT: return HTBOTTOMRIGHT;
+				case DRAG_MODE_RESIZE_BOTTOMLEFT: return HTBOTTOMLEFT;
+				case DRAG_MODE_NONE: return HTCLIENT;
```

Comment:
`DRAG_MODE_NONE` / `default` should return `DefWindowProcW(hWnd, uMsg, wParam, lParam);` instead, returning `HTCLIENT` will break resizing of normal (non-borderless) windows.